### DISCUSSION TITLE
Add back the "percentage" in `ArgonSongProgress`

### DISCUSSION THREAD

**Discussion by ikkue** [ppy/osu#22452] [Ideas] [answered]
I am referring to this percentage number in `DefaultSongProgress`.

![osu_2023-01-29_06-15-43](https://user-images.githubusercontent.com/11213135/215295642-a16ae825-3639-4d06-b67c-122c35ad2445.png)

It is currently not present both in the Figma and the latest implementation.

**Comment by bdach** [comment]
Turning this off was very deliberate (there was special code added to do just this).

Is this percentage ever actually useful to anybody?

**Reply by mk56-spn** [reply]
I learn which parts of the map are which by percentage not time ( and use this to target parts when wanting to quickly skim a replay) so it's not something totally out there IMO. 

An amicable solution in my view would be to extract it into its own component

**Reply by bdach** [reply]
I don't know about splitting it out. I'd prefer toggle on the component in this particular instance.

**Reply by ikkue** [reply]
> Is this percentage ever actually useful to anybody?

It's one of the many things that made me switch to lazer. When `LegacySongProgress` is not present in the skin, it is hard to judge how far into the map I am. `DefaultSongProgress` and `ArgonSongProgress` being linear as opposed to `LegacySongProgress` being circular makes it harder to tell, as generally, circles are easier to mentally divide than a long straight line at a quick glance. But a precise percentage number is still better than mentally dividing a circle.

**Reply by ItsShamed** [reply]
This is a simple change tho, I have a branch ready to be PRed. Just waiting for further feedback and opinions

**Reply by monochrome22** [reply]
Further feedback: Percentage is definitely useful, it feels like a more clear indicator of how long it's been and how much is left. You would think that seeing the exact amount of seconds left in the map would be the most clear, but apparently not. When I look at time left and it says 41 seconds, I just think "Is that a lot?" But with percentage, when I see 76%, it feels immediately obvious how much is left and how far I am. Percentage should definitely at least be a toggle, if not enabled by default.

**Reply by ItsShamed** [reply]
(PRing would worsen my case on #22489 lol, I'll wait a little more so that I don't spam PRs)

**Reply by peppy** [reply]
Based on this feedback, we may want to look at making a way to have it (optionally) displayed on the argon song progress. @arflyte cc in case you're interested.

**Comment by arflyte** [comment, marked as answer]
By default, the percentage is not there. The percentage was removed during mania skin design process (it made the area too busy). However, I had in mind to keep the percentage somewhere on the screen as an optional thing with a different design.
